app/scripts/metamask-controller.js

```diff
@@ -1018,15 +993,25 @@ export default class MetamaskController extends EventEmitter {
       removeIdentity: this.preferencesController.removeAddress.bind(
         this.preferencesController,
       ),
-      setAccountLabel: this.preferencesController.setAccountLabel.bind(
-        this.preferencesController,
-      ),
-      setSelectedAddress: this.preferencesController.setSelectedAddress.bind(
-        this.preferencesController,
-      ),
-      syncIdentities: this.preferencesController.syncAddresses.bind(
-        this.preferencesController,
-      ),
+      setAccountLabel: (address, label) => {
+        const accountToBeNamed =
+          this.accountsController.getAccountByAddress(address);
+        this.accountsController.setAccountName(accountToBeNamed.id, label);
```

Comment:
`accountToBeNamed` could be undefined, do we need to check it?

``` suggestion
        if (accountToBeNamed) {
          this.accountsController.setAccountName(accountToBeNamed.id, label);
        }
```

or throw an error?

``` suggestion
        if (accountToBeNamed) {
          this.accountsController.setAccountName(accountToBeNamed.id, label);
        } else {
          throw new Error("Could not find account");
        }
```

Reply:
Throwing an error sounds more appropriate. Done in [957a205](https://github.com/MetaMask/metamask-extension/pull/21437/commits/957a205fe9e0a800c0b8bbff8ac3e314456ace88)